generate-training-data.py

```diff
@@ -1,13 +1,38 @@
 import os
 import cv2
+import sys
 
-# write your code here to open your laptop's camera
-# and store images for rock paper scissors
+cap = cv2.VideoCapture(0)    
+cur_dir= os.getcwd()+'/'+'training_data'  #current directory 
+n = int(sys.argv[2])  #number of images to be captured
+folder_create = os.path.join(cur_dir, sys.argv[1]) #folder (with path) to be created 
+try:
+    os.mkdir(folder_create)  
+except FileExistsError:
+    pass                
 
-# I would highly recommend storing these images in folders stucture like below:
-# training_data\
-#           |-- empty\
-#           |-- rock\
-#           |-- paper\
-#           |-- scissors\
-# having a folder structure like this will make it easy for you to read the images while pre-processing
+img_counter=0
+while img_counter<n:
+	ret, frame = cap.read()
+
+	if not ret:
+	    print("failed to capture frame")
+	    cap.release()
+	    cv2.destroyAllWindows()
+
+	cv2.rectangle(frame, (100, 350), (450, 50), (0, 255, 0),15)  
+	cv2.imshow('capturing Gesture  : '+sys.argv[1],frame)
+	k = cv2.waitKey(1)
+	if k%256 == 27:#  if ESC pressed
+		print("Escape hit, closing...")
+		cap.release()
+		cv2.destroyAllWindows()
+	elif k%256 == 32: #if SPACE pressed
```

Comment:
This is a very good approach to capture images only when "SPACE" is pressed, but it becomes infeasible to keep pressing the key or pressing and holding it for a long time.

A better approach would be a toggle on the key-press.